### PR TITLE
Ci improve clippy 2

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

**Comment by cryptonemo** [comment]
FYI, I have a minor update incoming for the benches

**Comment by cryptonemo** [comment]
Builds on #1461

File: fil-proofs-param/src/bin/paramfetch.rs [hunk 1]
```diff
@@ -368,29 +370,29 @@ pub fn main() {
...
-            }
-            ipget_path
+
+        path
+    } else {
+        let ipget_version = cli
+            .ipget_version
+            .unwrap_or_else(|| DEFAULT_IPGET_VERSION.to_string());
+        let tmp_path = get_ipget_path(&ipget_version);
+        let path = Path::new(&tmp_path);
```
[the remark this answers is not in the document]
Comment (cryptonemo) [reply fil-proofs-param/src/bin/paramfetch.rs:389]: Give it a shot.  In general clippy complained about almost every usage of PathBuf.  I don't recall that issue in particular, but worth adjusting if you can find a cleaner way.

File: filecoin-hashers/src/types.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ pub trait Domain:
     + Element
     + StdHash
 {
-    fn into_bytes(&self) -> Vec<u8>;
+    fn into_bytes(self) -> Vec<u8>;
```
[the remark this answers is not in the document]
Comment (cryptonemo) [reply filecoin-hashers/src/types.rs:37]: This was very specifically clippy directed.  Give it a try.

File: fil-proofs-param/src/bin/paramfetch.rs [hunk 3]
```diff
@@ -114,12 +114,14 @@ fn download_ipget(version: &str, verbose: bool) -> Result<()> {
...
     trace!("writing downloaded file to: {}", write_path);
     let mut writer = File::create(&write_path).expect("failed to create file");
-    if verbose && size.is_some() {
-        let mut resp = FetchProgress::new(resp, size.unwrap());
-        copy(&mut resp, &mut writer).expect("failed to write download to file");
+    if verbose {
+        if let Some(size) = size {
+            let mut resp = FetchProgress::new(resp, size);
+            copy(&mut resp, &mut writer).expect("failed to write download to file");
+        }
```
[the remark this answers is not in the document]
Comment (vmx) [reply fil-proofs-param/src/bin/paramfetch.rs:124]: What I propose soesn't work, as `FetchProgress` is a different type. Wow, that was confusing.

What about renaming that second `resp` to `resp_with_progress`? I don't have a strong opinion about that, so feel free to keep the current code.

[the remark this answers is not in the document]
Comment (cryptonemo) [reply fil-proofs-param/src/bin/paramfetch.rs:124]: Any and all adjustments welcome.  Unfortunately, clippy is one of these things that we're all going to have to manage over time.  PRs like this ideally won't exist going forward (i.e. specifically fixing clippy issues across 50+ files).

File: filecoin-hashers/src/types.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vmx) [reply filecoin-hashers/src/types.rs]: Done.

**Comment by vmx** [comment]
I've pushed the changes I proposed.

**Review by vmx — APPROVED** [review]

**Review by vmx — APPROVED** [review]

**Comment by cryptonemo** [comment]
@vmx This PR was replaced by #1465

**Comment by cryptonemo** [comment]
Closing in favor of #1465

**Pull request closed without merging**